catalog/spatial/wps/spatial-wps-endpoint/src/main/java/org/codice/ddf/spatial/ogc/wps/process/endpoint/WpsObjectConverter.java

```diff
@@ -0,0 +1,819 @@
+      }
+      if (dataDesc instanceof DataDescriptionGroup
+          && !((DataDescriptionGroup) dataDesc).getDataDescriptions().isEmpty()) {
+        inputDescriptionType.setInput(
+            createInputDescriptions(((DataDescriptionGroup) dataDesc).getDataDescriptions()));
+      } else {
+        inputDescriptionType.setDataDescription(createDataDescriptionType(dataDesc));
+      }
+      inputDescriptionTypes.add(inputDescriptionType);
+    }
+    return inputDescriptionTypes;
+  }
+
+  public List<ProcessSummaryType> createProcessSummary(ProcessRepository processesRepository) {
+    List<ProcessSummaryType> processSummaryTypes = new ArrayList<>();
+    for (Process process : processesRepository.getProcesses()) {
+      try {
+
+        ProcessSummaryType processSummaryType = wpsObjectFactory.createProcessSummaryType();
+        processSummaryType.setJobControlOptions(getJobControlOptions(process.getOperations()));
+        processSummaryType.setProcessVersion(process.getVersion());
+        processSummaryType.setIdentifier(createCodeType(process.getId()));
+        processSummaryType.setTitle(createLanguageStrings(process.getTitle()));
+        processSummaryType.setAbstract(createLanguageStrings(process.getDescription()));
+        processSummaryType.setMetadata(createMetadata(process.getMetadata()));
+        if (process.getMetadata() != null) {
+          processSummaryType.setKeywords(createKeywords(process.getMetadata().getKeywords()));
+        }
+        processSummaryType.setOutputTransmission(
+            createDataTransmissionType(process.getExecutionDescription().getTransmissionModes()));
+        processSummaryTypes.add(processSummaryType);
+      } catch (RuntimeException e) {
+        LOGGER.debug("Error getting process summary from process: {}", process.getId(), e);
+      }
+    }
+    return processSummaryTypes;
+  }
+
+  public List<ProcessOffering> createProcessOfferings(ProcessRepository processRepository) {
+    List<ProcessOffering> processOfferings = new ArrayList<>();
+
+    for (Process process : processRepository.getProcesses()) {
+      try {
+        processOfferings.add(createProcessOffering(process));
+      } catch (RuntimeException e) {
+        LOGGER.debug("Error creating process offering", e);
+      }
+    }
+    return processOfferings;
+  }
+
+  public List<ProcessOffering> createProcessOfferings(
+      ProcessRepository processRepository, Set<String> processIds) {
+    return processIds
+        .stream()
+        .map(
+            processId -> {
+              try {
+                Optional<Process> process = processRepository.getProcess(processId);
+                if (process.isPresent()) {
+                  return createProcessOffering(process.get());
+                }
+              } catch (RuntimeException e) {
+                LOGGER.debug("Error getting algorithm {}", processId, e);
+              }
+              return null;
```

Comment:
❗️ I would recommend against intermixing nulls/optionals - instead use the optional.map to continue the null conditional logic and do the overall Optional::isPresent + Optional::get at the outer streaming layer